plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/errormarker/FordiacMarkerHelper.java

```diff
@@ -130,15 +132,22 @@ public static Map<String, Object> getDiagnosticAttributes(final Diagnostic diagn
 			final List<?> data = diagnostic.getData();
 			if (data != null && data.size() >= 5) {
 				if (data.get(4) != null) {
-					return Map.of(IMarker.LOCATION, data.get(1), FordiacErrorMarker.TARGET_URI, data.get(2),
-							FordiacErrorMarker.TARGET_TYPE, data.get(3), FordiacErrorMarker.TARGET_FEATURE,
-							data.get(4));
+					return Map.of(FordiacErrorMarker.CODE, Integer.valueOf(diagnostic.getCode()),
```

Comment:
can you make a java class or record for this to wrap the data ?
This code is really hard to read.

Reply:
I am also somewhat unhappy with the situation. Unfortunately, I am quite restricted by both the Eclipse platform and Xtext on what types I can use here. The original `Diagnostic.getData()` array must only contain exactly one `EObject` (i.e., the target) and the rest must be plain `String`s to support conversion of the `Diagnostic` to an `org.eclipse.xtext.validation.Issue`. The marker attributes, on the other hand, only support `String`, `Integer`, or `Boolean` as values.

Nevertleless, I tried improving the formatting to (hopefully) make it a bit easier to read.